apps/aehyperchains/test/connectors/aehc_chain_sim_connector.erl

```diff
@@ -46,22 +46,22 @@ get_block_by_hash(Hash) ->
 
 -record(state, { pid::pid(), height = 0::non_neg_integer() }).
 
-init([]) ->
+init(_Args) ->
     process_flag(trap_exit, true),
     true = aec_events:subscribe(top_changed),
     {ok, Pid} = aec_chain_sim:start(#{ simulator => parent_chain }),
     lager:info("Parent chain's connector ~p is attached: ~p", [?MODULE, Pid]),
     {ok, #state{ pid = Pid }}.
 
-handle_call({send_tx, Payload}, _From, State) ->
+handle_call({send_tx, Delegate, _Commitment, _PoGF}, _From, State) ->
     %% The current validator credentials;
     %% Requested transaction by hash from a simulator's block should satisfy the origin of validator;
-    {ok, Pub} = aec_keys:pubkey(),
     {ok, PrivKey} = aec_keys:sign_privkey(),
-    SenderId = aeser_id:create(account, Pub),
     %% The main intention of this call is to emulate post action with signed payload from delegate;
     %% Fee, nonce, ttl and amount fields have decorated nature;
-    {ok, Tx} = aec_spend_tx:new(#{ sender_id => SenderId, recipient_id => SenderId, amount => 1,
+    Header = aehc_commitment_header:new(<<"D1">>, <<"BLOCK 1">>),
+    Payload = aehc_commitment_header:hash(Header),
+    {ok, Tx} = aec_spend_tx:new(#{ sender_id => Delegate, recipient_id => Delegate, amount => 1,
                                         fee => 5, nonce => 1, payload => Payload, ttl => 0 }),
```

Comment:
You are sending the commitment ID instead of the contents.

Reply:
+1